python/whylogs/api/logger/experimental/logger/actor/thread_actor.py

```diff
@@ -0,0 +1,91 @@
+import queue
+import threading as th
+from typing import Generic, List, Optional, TypeVar
+
+from whylogs.api.logger.experimental.logger.actor.actor import (
+    Actor,
+    QueueConfig,
+    QueueWrapper,
+)
+
+DefaultQueueWrapperType = TypeVar("DefaultQueueWrapperType")
+
+
+class ThreadQueueWrapper(QueueWrapper, Generic[DefaultQueueWrapperType]):
+    """
+    Implementation of QueueWrapper sufficient for use in the threaded actor.
+    """
+
+    def __init__(self) -> None:
+        self._queue: queue.Queue[DefaultQueueWrapperType] = queue.Queue(100_000)
+
+    def send(self, message: DefaultQueueWrapperType, timeout: float = 0.1) -> None:
+        self._queue.put(message, timeout=timeout)
+
+    def send_many(self, messages: List[DefaultQueueWrapperType], timeout: float = 0.1) -> None:
+        for message in messages:
+            self._queue.put(message, timeout=timeout)
+
+    def get(self, timeout: float = 0.1) -> Optional[DefaultQueueWrapperType]:
+        return self._queue.get(timeout=timeout)
+
+    def get_many(self, timeout: float = 0.1, max: Optional[int] = None) -> List[DefaultQueueWrapperType]:
+        if max is None or max < 1:
+            return []
+
+        messages: List[DefaultQueueWrapperType] = []
+
+        while True:
+            if len(messages) >= max:
+                break
+
+            try:
+                messages.append(self._queue.get(timeout=timeout))
+            except queue.Empty:
+                if len(messages) == 0:
+                    raise queue.Empty
+                else:
+                    break
+
+        return messages
+
+    def size(self) -> int:
+        return self._queue.qsize()
+
+
+ThreadMessageType = TypeVar("ThreadMessageType")
```

Comment:
Shouldn't it be right next to the place its actually used? Nothing else would really care about this generic class token right? If this were a real language it wouldn't even exist, it would just be `T` in the class def.